### PR TITLE
[core]Add `IF_TYPE_OF` to enable CC for `IfTypeOf` class

### PR DESCRIPTION
**Pull request by praasz**
### Details:
- Integrate `ov::element::IfTypeOf` class with conditional compilation to remove unused precision from operators.
- Add macro `IF_TYPE_OF` to warp class `ov::element::IfTypeOf`  end enable CC features.
- Update `ccheader.py` to created PP symbols for template parameter list required by `IF_TYPE_OF` macro.
- Use new macro in Convert operator as example.

### Tickets:
 - [CVS-124984](https://jira.devtools.intel.com/browse/CVS-124984)

### CODE REVIEW

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/common/util/include/openvino/util/pp.hpp [hunk 1]
```diff
@@ -48,3 +48,12 @@
...
 #define OV_PP_IS_ENABLED2(arg1_or_junk) OV_PP_SECOND_ARG(arg1_or_junk 1, 0)
+
+// Empty symbol
+#define OV_PP_EMPTY
+
+// Eats inputs
+#define OV_PP_EAT(...)
+
+// Expand input
+#define OV_PP_EXP(...) __VA_ARGS__
```
[the remark this answers is not in the document]
Comment (praasz) [reply src/common/util/include/openvino/util/pp.hpp:59]: Name has been used, already.

[the remark this answers is not in the document]
Comment (t-jankowski) [reply src/common/util/include/openvino/util/pp.hpp:59]: Can the other one used here?

[the remark this answers is not in the document]
Comment (praasz) [reply src/common/util/include/openvino/util/pp.hpp]: No there is only one `OV_PP_EXPAND`

File: src/core/shape_inference/include/element_visitor.hpp [hunk 2]
```diff
@@ -120,3 +148,37 @@ struct NotSupported {
 };
 }  // namespace element
 }  // namespace ov
+
+// Return ov::elements as parameter list e.g. OV_PP_ET_LIST(f16, i32) -> f16, i32
+#define OV_PP_ET_LIST(...) OV_PP_EMPTY __VA_ARGS__
+
+// Helpers to implement eat or expand if symbol exists
```
Comment (vurusovs) [inline src/core/shape_inference/include/element_visitor.hpp:155]: eat -> ignore?

File: src/core/src/op/convert.cpp [hunk 3]
```diff
@@ -26,12 +26,14 @@ struct Evaluate : public element::NoAction<bool> {
...
     static result_type visit(const Tensor& arg, Tensor& out, const size_t count) {
         using namespace ov::element;
-        return IfTypeOf<CONVERT_ET_LIST>::apply<EvalByOutputType<is_lp_type(ET)>>(
-            out.get_element_type(),
-            reinterpret_cast<const TI*>(arg.data()),
-            out,
-            count,
-            ET);
+        return IF_TYPE_OF(Convert_out,
+                          CONVERT_ET_LIST,
```
Comment (vurusovs) [inline src/core/src/op/convert.cpp:30]: Is it expected that `CONVERT_ET_LIST` not wrapped with `OV_PP_ET_LIST`. Could you explain in detail how `OV_PP_ET_LIST` works?

File: src/common/util/include/openvino/util/pp.hpp [hunk 4]
```diff
@@ -48,3 +48,9 @@
 
 // Return second argument from possible sequences {1, 0}, {0, 1, 0}
 #define OV_PP_IS_ENABLED2(arg1_or_junk) OV_PP_SECOND_ARG(arg1_or_junk 1, 0)
+
+// Empty symbol
+#define OV_PP_EMPTY
```
[the remark this answers is not in the document]
Comment (praasz) [reply src/common/util/include/openvino/util/pp.hpp:53]: It looks not required for now, removed.

File: src/core/shape_inference/include/element_visitor.hpp [hunk 5]
```diff
@@ -8,10 +8,17 @@
 
 #include "openvino/core/except.hpp"
 #include "openvino/core/type/element_type.hpp"
+#include "openvino/itt.hpp"
 
 namespace ov {
-namespace element {
+namespace itt {
+namespace domains {
+OV_ITT_DOMAIN(ov_eval);
```
Comment (vurusovs) [inline src/core/shape_inference/include/element_visitor.hpp:16]: is this domain used somewhere?

Comment (praasz) [reply src/core/shape_inference/include/element_visitor.hpp:16]: Yes is used to generate `IF_TYPE_OF` macro during CC analysis.

File: src/core/shape_inference/include/element_visitor.hpp [hunk 6]
```diff
@@ -71,6 +85,20 @@ struct IfTypeOf<ET, Others...> {
         return (et == ET) ? Visitor::template visit<ET>(std::forward<Args>(args)...)
                           : IfTypeOf<Others...>::template apply<Visitor>(et, std::forward<Args>(args)...);
     }
+
+#if defined(SELECTIVE_BUILD_ANALYZER)
```
Comment (t-jankowski) [inline src/core/shape_inference/include/element_visitor.hpp:89]: Different style than above
```
#if defined SELECTIVE_BUILD_ANALYZER
```

**Review by t-jankowski — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**